### PR TITLE
Buidl UI for search feature

### LINKED ISSUE #5796: Update Site search module with design
The scope of this task is to update https://github.com/gitcoinco/web/pull/5797 with the following design
search

and to add the tabs that are in the design to the site functionality

### PR DESCRIPTION
**Pull request by iamonuwa**
##### Description
Build UI according to figma design

##### Refers/Fixes
Fixes #5796 

##### Testing

Click the search on navbar.

### CODE REVIEW

**Comment by iamonuwa** [comment]
screen-capture-_6_

File: app/dashboard/templates/shared/nav_auth.html [hunk 1]
```diff
@@ -107,4 +107,4 @@
     <i class="fab fa-github"></i><span class="nav-link__text">{% trans "Log in" %}</span>
   </a>
 {% endif %}
-<div id=search_container></div>
+<!-- <div id=search_container></div> -->
```
Comment (thelostone-mc) [inline app/dashboard/templates/shared/nav_auth.html]: remove this ?

File: app/dashboard/templates/shared/search.html [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                      <span>[[ result.source_type | capitalize ]]</span>
+                    </div>
+                  </div>
+                </a>
+              </li>
+            </ul>
+          </div>
+      </div>
+    </div>
+</div>
```
Comment (thelostone-mc) [inline app/dashboard/templates/shared/search.html]: ^ re-indent with 2 spaces

**Comment by owocki** [comment]
i've run create_search_results on my local.. and i still cant' get this functionality to work. it never makes any HTTP request to make search queries or shows any results..

**Comment by iamonuwa** [comment]
What error do you get?

**Comment by owocki** [comment]
no error, just nothing shows up or changes in the search bar

**Comment by iamonuwa** [comment]
> no error, just nothing shows up or changes in the search bar

i'll take a look now

**Comment by iamonuwa** [comment]
cc: @owocki 
I was unable to reproduce the error. it works fine for me.

screen-capture-_10_

**Comment by iamonuwa** [comment]
cc @owocki 
I was unable to reproduce the error. It works fine for me.

screen-capture-_11_

**Comment by owocki** [comment]
@iamonuwa is that a Search Result from create_search_results?  r u able to generate lots of search results?

**Comment by owocki** [comment]
oh wait i got the profile to work !!! by searching my handle name :)

does it work for all tabs for u?

**Comment by owocki** [comment]
@iamonuwa any idea when the API returns 100s of results.. why does it not show in the UI https://bits.owocki.com/4guxpA94

**Comment by owocki** [comment]
two other things that are messed up on this PR (beyond just not being able to load results)
- search bar isnt mobile responsive
- this breaks the bounty explorer at `/explorer`

**Comment by iamonuwa** [comment]
@owocki, I'll make this responsive more. But I'm only using your backend update. I did not make changes to it.

**Comment by owocki** [comment]
ok well my PR doesnt break the /explorer page, and the results load in all scenarios on my PR... those issues will need to be fixed

**Comment by thelostone-mc** [comment]
@iamonuwa let us know when these are fixed and we can get it reviewed + merged

File: app/assets/v2/css/search.css [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+  --badge-blue-text: #6587ae;
+}
+
+.navbar-nav .nav-link.search__icon {
+  position: relative;
+  font-size: 18px;
+  color: white;
+}
+
+.nav-line {
```
Comment (octavioamu) [inline app/assets/v2/css/search.css]: Don't restyle bootstrap general classes (or gitcoin general classes) unless you want to change the visual in all the site.

File: app/assets/v2/css/search.css [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+  display: block;
+  width: 2.375rem;
+  height: 2.375rem;
+  line-height: 2.375rem;
+  text-align: center;
+  pointer-events: none;
+  color: #aaa;
+}
+
+.nav-item {
```
Comment (octavioamu) [inline app/assets/v2/css/search.css]: general calsse will set all menu to 10px and bold

File: app/assets/v2/css/search.css [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+  font-size: 10px;
+  font-weight: 800;
+}
+
+.search-result__description .tag {
+  display: inline-block;
+}
+
+.tag {
```
Comment (octavioamu) [inline app/assets/v2/css/search.css]: will break tag class used on many places, like explorer, bounty details, user explorer... etc

File: app/assets/v2/css/search.css [hunk 6]
```diff
@@ -0,0 +1,170 @@
+:root {
```
Comment (octavioamu) [inline app/assets/v2/css/search.css]: var defined as root by not used and seems only related to search not root, maybe something for colors.css instead ?

**Comment by thelostone-mc** [comment]
@iamonuwa any update on this?

**Comment by iamonuwa** [comment]
I'm on it

**Comment by iamonuwa** [comment]
> ok well my PR doesnt break the /explorer page, and the results load in all scenarios on my PR... those issues will need to be fixed

what do you mean by break explorer page?

**Comment by iamonuwa** [comment]
Fixed this PR.

**Comment by danlipert** [comment]
@iamonuwa any updates?

**Comment by thelostone-mc** [comment]
@iamonuwa ^ could you give us an update ? Else this PR will end up getting stale

**Comment by iamonuwa** [comment]
I've resolved all the issues raised. See my last commit

**Comment by iamonuwa** [comment]
@thelostone-mc @danlipert @octavioamu @owocki fixed all changes.

File: app/search/migrations/0001_initial.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
+# Generated by Django 2.2.4 on 2020-01-18 15:31
```
Comment (danlipert) [inline app/search/migrations/0001_initial.py:1]: Where did this file come from?

Comment (iamonuwa) [reply app/search/migrations/0001_initial.py:1]: Weird, let me take a look at that

Comment (iamonuwa) [reply app/search/migrations/0001_initial.py:1]: Weird, let me take a look at that

**Comment by thelostone-mc** [comment]
Closing this out and redoing it over at https://github.com/gitcoinco/web/pull/5835!
Thanks @iamonuwa for getting the base ready for this

**Pull request closed without merging**